### ISSUE TITLE
Defined type instances are reported as not inheriting from Object

### ISSUE THREAD

**Issue by axelcostaspena** [Microsoft/TypeScript#5111]
tsc can't compile this code complaining about 
`Error:(14, 7) TS2345: Argument of type 'typeof A' is not assignable to parameter of type 'ObjectConstructor'.
 Property 'getPrototypeOf' is missing in type 'typeof A'.`

    class A {
    
    }
    class B {
        public Foo(anA: typeof A):void {
    
        }
        public Bar(anA: typeof Object):void {
    
        }
    }
    
    var b = new B();
    b.Foo(A);
    b.Bar(A); // <- error here

Not even this ugly workaround works:

    class A extends Object {
    
    }

**Comment by kitsonk** [comment]
What specifically are you trying to guard against in `Bar`?  Because `typeof Object` is retrieving the structural typing of the Object constructor for JavaScript, which `A` is not of that type.  Are you trying to type guard against any sort of constructor function?  If so, you want something like this:

```typescript
interface GenericClass {
    new (...args: any[]): any;
}

class A {

}

class B {
  public Bar(anA: GenericClass): void { }
}

var b = new B();
b.Bar(A);
```

**Comment by RyanCavanaugh** [comment]
@kitsonk is right as usual. The type `typeof Object` has members like `getPrototypeOf`, `defineProperty`, `freeze`, etc. which are not present on `A`. It seems like you're trying to represent some other pattern; outlining what the expected constraint on `Bar` is would help us clarify the right thing to write.

**Comment by axelcostaspena** [comment]
Yes, @kitsonk is true. I was trying to type my parameter as any newable type. Sorry for the fake issue.
